api/admin/controller.py

```diff
@@ -2240,42 +2330,102 @@ def admin_auth_service(self, protocol):
     def individual_admins(self):
         if flask.request.method == 'GET':
             admins = []
-            admins_with_password = Admin.with_password(self._db)
-            if admins_with_password.count() != 0:
-                admins=[dict(email=admin.email) for admin in admins_with_password]
+            for admin in self._db.query(Admin):
+                roles = []
+                for role in admin.roles:
+                    if role.library:
+                        if not flask.request.admin or not flask.request.admin.is_librarian(role.library):
+                            continue
+                        roles.append(dict(role=role.role, library=role.library.short_name))
+                    else:
+                        roles.append(dict(role=role.role))
+                admins.append(dict(email=admin.email, roles=roles))
```

Comment:
I think that's correct, because we'd like a library manager to be able to add them to their library even if they don't have any other roles that are visible to that manager yet.